[build] Need to update cmake_minimum_required() to 3.17

**Describe the bug**
Iam trying to build multipass from source on my local machine . It is failing at cmake step. 

**To Reproduce**
How, and what happened?

1 . git clone https://github.com/canonical/multipass.git
2. cd multipass
3. apt install devscripts equivs
4. mk-build-deps -s sudo -i

5. cd multipass
6. git submodule update --init --recursive
7. mkdir build
8. cd build
9. cmake ../
10. make


**Expected behavior**
I was expecting the build to be successful

**Additional info**
 - OS: ubuntu 20.04 

**Additional context**
multipass/build$ cmake ../
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1f")
-- Populating grpc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/arrka/tools/multipass/multipass/build/_deps/grpc-subbuild
Scanning dependencies of target grpc-populate
[ 11%] Creating directories for 'grpc-populate'
[ 22%] Performing download step (git clone) for 'grpc-populate'
Cloning into 'grpc-src'...
remote: Enumerating objects: 121739, done.
remote: Counting objects: 100% (121739/121739), done.
remote: Compressing objects: 100% (58726/58726), done.
remote: Total 121739 (delta 89488), reused 81563 (delta 56905), pack-reused 0
Receiving objects: 100% (121739/121739), 76.67 MiB | 34.57 MiB/s, done.
Resolving deltas: 100% (89488/89488), done.
Note: switching to 'ba8e7f72'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at ba8e7f72a Update patches to work with newer grpc code
error: pathspec 'GIT_SUBMODULES_RECURSE' did not match any file(s) known to git
error: pathspec 'false' did not match any file(s) known to git
CMake Error at grpc-subbuild/grpc-populate-prefix/tmp/grpc-populate-gitclone.cmake:52 (message):
  Failed to update submodules in:
  '/home/myself/tools/multipass/multipass/build/_deps/grpc-src'


make[2]: *** [CMakeFiles/grpc-populate.dir/build.make:92: grpc-populate-prefix/src/grpc-populate-stamp/grpc-populate-download] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/grpc-populate.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
CMake Error at /usr/share/cmake-3.16/Modules/FetchContent.cmake:915 (message):
  Build step for grpc failed: 2
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FetchContent.cmake:1006 (__FetchContent_directPopulate)
  /usr/share/cmake-3.16/Modules/FetchContent.cmake:1047 (FetchContent_Populate)
  3rd-party/CMakeLists.txt:14 (FetchContent_MakeAvailable)


-- Configuring incomplete, errors occurred!
See also "/home/myself/tools/multipass/multipass/build/CMakeFiles/CMakeOutput.log".
myself@host:~/tools/multipass/multipass/build$

@townsend2010 Thanks for the inputs. The above issue is resolved after upgrading cmake. I further faced issue due to extra ; at following places. After removing the ;  at following places iam able to build successfully

/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:615:52: error: extra ‘;’ [-Werror=pedantic]
  615 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(Descriptor, 136);
  
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:992:56: error: extra ‘;’ [-Werror=pedantic]
  992 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(FieldDescriptor, 72);
  
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1073:56: error: extra ‘;’ [-Werror=pedantic]
 1073 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(OneofDescriptor, 40);
      |                                                        ^
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1244:55: error: extra ‘;’ [-Werror=pedantic]
 1244 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(EnumDescriptor, 72);
      |                                                       ^
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1329:60: error: extra ‘;’ [-Werror=pedantic]
 1329 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(EnumValueDescriptor, 32);
      |                                                            ^
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1412:58: error: extra ‘;’ [-Werror=pedantic]
 1412 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(ServiceDescriptor, 48);
      |                                                          ^
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1501:57: error: extra ‘;’ [-Werror=pedantic]
 1501 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(MethodDescriptor, 64);
      |                                                         ^
/build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/descriptor.h:1705:56: error: extra ‘;’ [-Werror=pedantic]
 1705 | PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(FileDescriptor, 144); 
 
 /build/_deps/grpc-src/third_party/protobuf/src/google/protobuf/parse_context.h:611:2: error: extra ‘;’ [-Werror=pedantic]
  611 | };

--- a/3rd-party/CMakeLists.txt
+++ b/3rd-party/CMakeLists.txt
@@ -76,7 +76,7 @@ target_link_libraries(gRPC INTERFACE
   zlibstatic)
 
 if (NOT MSVC)
-  target_compile_options(gRPC INTERFACE "-Wno-unused-parameter" "-Wno-non-virtual-dtor")
+  target_compile_options(gRPC INTERFACE "-Wno-unused-parameter" "-Wno-non-virtual-dtor" "-Wno-pedantic")
 endif ()
 
 # YAML C++

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.17)
 cmake_policy(SET CMP0079 NEW) # Allow target_link_libraries() in subdirs
 
 if (APPLE)

